cpp/src/parquet/column_reader.cc

```diff
@@ -1369,6 +1369,24 @@ class TypedRecordReader : public TypedColumnReaderImpl<DType>,
     return bytes_for_values;
   }
 
+  const uint8_t* ReadDictionary(int32_t* dictionary_length) override {
+    if (this->current_decoder_ == nullptr && !this->HasNextInternal()) {
+      dictionary_length = 0;
+      return nullptr;
+    }
+    // Verify the current data page is dictionary encoded.
+    if (this->current_encoding_ != Encoding::RLE_DICTIONARY) {
```

Comment:
doesn't this need to be plain dictionary also?

Reply:
or actually, I guess PLAIN?

Reply:
Perhaps adding a comment to clear the confusion?